### PR TITLE
Remove FORMAT <..> backwards compatibility options from COPY

### LINKED ISSUE #9905: [DISCUSS] Remove `FORMAT <..>` backwards compatibility options from COPY
### Is your feature request related to a problem or challenge?

@tinfoil-knight  added backwards compatibility to the COPY command format options in version 37.0.0 via  https://github.com/apache/arrow-datafusion/pull/9744

So now this format is supported (similar to duckdb)

```sql
COPY (select * from (values (1))) to 'test_files/scratch/copy/'
OPTIONS (format json, compression gzip);
```

However, the newer more consistent syntax looks like
```sql
COPY (select * from (values (1))) to 'test_files/scratch/copy/'
STORED AS JSON
OPTIONS (compression gzip);
```

@metesynnada  asked https://github.com/apache/arrow-datafusion/pull/9753#pullrequestreview-1970055037 if we should phase out the old (`format json`) syntax

### Describe the solution you'd like

Decide if  filing this ticket to discuss

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tinfoil-knight**
## Which issue does this PR close?

Closes #9905 .

## Rationale for this change

Please see the issue description.

## What changes are included in this PR?

- Reverts #9753 
- Reverts #9744

## Are these changes tested?

Tests have been updated as the backward compatibility for `format` has been removed now.

## Are there any user-facing changes?

Yes. Some related doc updates were done in  #9931. I've removed the leftover references to the old `(format ...)` options syntax from docs & examples.

### CODE REVIEW

**Review by alamb — APPROVED** [review]
Thank you @tinfoil-knight  🙏 

cc @metesynnada @devinjdangelo @berkaysynnada

File: datafusion/sql/src/statement.rs [hunk 1]
```diff
@@ -850,7 +850,7 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                     return plan_err!("Unsupported Value in COPY statement {}", value);
                 }
             };
-            if !(key.contains('.') || key == "format") {
+            if !(&key.contains('.')) {
```
Comment (devinjdangelo) [inline datafusion/sql/src/statement.rs:853]: :+1:

**Review by devinjdangelo — APPROVED** [review]
Thank you @tinfoil-knight !

**Comment by alamb** [comment]
Thanks again @tinfoil-knight

**Pull request merged**